machineconfiguration/v1/types_machineosbuild.go

```diff
@@ -104,16 +104,16 @@ type MachineOSBuildStatus struct {
 
 // MachineOSBuilderReference describes which ImageBuilder backend to use for this build
 // +union
-// +kubebuilder:validation:XValidation:rule="has(self.imageBuilderType) && self.imageBuilderType == 'PodImageBuilder' ?  has(self.buildPod) : !has(self.buildPod)",message="buildPod is required when imageBuilderType is PodImageBuilder, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.imageBuilderType) && self.imageBuilderType == 'JobImageBuilder' ?  has(self.jobImageBuilderStatus) : !has(self.jobImageBuilderStatus)",message="jobImageBuilderStatus is required when imageBuilderType is JobImageBuilder, and forbidden otherwise"
 type MachineOSBuilderReference struct {
 	// ImageBuilderType describes the image builder set in the MachineOSConfig
 	// +unionDiscriminator
 	ImageBuilderType MachineOSImageBuilderType `json:"imageBuilderType"`
 
-	// relatedObjects is a list of objects that are related to the build process.
+	// JobImageBuilderStatus describes the status of the Job doing the image build
 	// +unionMember
 	// +optional
-	PodImageBuilder *ObjectReference `json:"buildPod,omitempty"`
+	JobImageBuilderStatus *ObjectReference `json:"jobImageBuilderStatus,omitempty"`
```

Comment:
I'm not sure that `BuilderStatus` makes sense here because this is mostly just a reference to the build executor which doesn't include any kind of status message (to my knowledge). Maybe something like `JobImageBuilderRef` or something like that would be more appropriate?

Reply:
I think the comment above this field might need an update to reflect that it is a reference as well. 

Stepping back a bit, I'm guessing the reason we are using a union here is we could have different builders, and they could need different types(other than `ObjectReference`) to represent their reference? If we are always going to use `ObjectReference` for multiple kind of builders, then maybe we don't even need to have to a union discriminator? It could be just two fields within this struct as Zack suggested: type and ref.

Reply:
Those values are arguably common to all image builders. There could be some additional information that may be desired specific to each type of image builder, but we can add that onto this later as a specific type.

Reply:
And taking a step back from this, if the `ObjectReference` includes GVK (Group Version Kind) information, we could potentially eliminate the `ImageBuilderType` field here since I see that as being primarily user-facing.

Reply:
Ok, so instead of having individual ImageBuilderRef's, I'll just make it one optional field here to encompass all potential object references.